src/common/types/profile-derived.js

```diff
@@ -8,15 +8,44 @@ import type { MarkerPayload } from './profile';
 
 export type IndexIntoFuncStackTable = number;
 
+/**
+ * Contains a table of stack information that is unique to the function as opposed to
+ * being unique to the frame. There can be multiple frames for a single C++ function.
+ * Using stacks as opposed to funcStacks can cause duplicated functions in reports
+ * like the call tree.
+ *
+ * For example:
+ *
+ *            stack1 (funcA)                             funcStack1 (funcA)
+ *                 |                                            |
+ *                 v                                            v
+ *            stack2 (funcB)         stackTable to       funcStack2 (funcB)
+ *                 |                funcStackTable              |
+ *                 v                      ->                    v
+ *            stack3 (funcC)                             funcStack3 (funcC)
+ *            /            \                                    |
+ *           V              V                                   v
+ *    stack4 (funcD)     stack5 (funcD)                  funcStack4 (funcD)
+ *         |                  |                          /               \
+ *         v                  V                         V                 V
+ *    stack6 (funcE)     stack7 (funcF)       funcStack5 (funcE)     funcStack6 (funcF)
```

Comment:
Nice drawing. Would be fun to have a diamond in here as well, but it would probably clutter it up too much.